src/domain/customimage/CustomImage.java

```diff
@@ -15,10 +18,29 @@
     private final PixelReader reader;
     private final BufferedImage bufferedImage;
     private final Format format;
+    private int[][] redMatrix;
+    private int[][] greenMatrix;
+    private int[][] blueMatrix;
     private List<Pixel> pixelList;
 
-    public CustomImage(Image image, String formatString){
+    public CustomImage(Image image, String formatString) {
         this(SwingFXUtils.fromFXImage(image, null), formatString);
+
+        this.redMatrix = MatrixService.toChannelMatrix(image, (x, y) -> image.getPixelReader().getColor(x, y).getRed());
+        this.greenMatrix = MatrixService.toChannelMatrix(image, (x, y) -> image.getPixelReader().getColor(x, y).getGreen());
+        this.blueMatrix = MatrixService.toChannelMatrix(image, (x, y) -> image.getPixelReader().getColor(x, y).getBlue());
```

Comment:
Ah, por esto eran estaticos? Creo que seria mas piola que el propio constructor llamara al provider y le pidiera el servicio, asi no tenemos necesidad de hacer ese metodo estático. Al fin y al cabo, ya es un singleton

Reply:
Ah no, bueno, no es un singleton pero no tiene atributos ni nada que lo diferencie

Reply:
Yo por ahora lo dejo, creo que puede servir. Meterlo dentro de la custom image podria obligarnos a cambiar unas cuantas cosas, creo. De ultima lo anoto en una tarjetita de Trello como un to refactor a futuro, si vemos que no da problemas